### ISSUE TITLE
Scrolling page down via <Space> scrolls too much

### ISSUE THREAD

**Issue by now** [akhodakivskiy/VimFx#763]
Steps to reproduce:

1. Press `<space>` to scroll the page down

Expected result: Scroll one page full downwards.

Actual result: Scrolls about one and a half pages downwards.

VimFx version: 0.17.1
Firefox version: 47
OS: Mac OS
Keyboard language: en-US (Dvorak)
Worked before: Yes
Example URL(s): https://www.kernel.org/category/about.html

**Comment by d125q** [comment]
Scrolls the same amount as PgUp here.

**Comment by now** [comment]
Both page-scrolling commands scroll two page up/page down’s worth of screen in my set-up.  Changing <Space> to use the half-page scrolling function solves this for me.

**Comment by lydell** [comment]
I can't reproduce either. Please test in a new Firefox profile.

VimFx version: 0.17.1
Firefox version: 47.0
OS: Ubuntu 16.04
Keyboard language: en-US

**Comment by now** [comment]
Enabling *Use Smooth Scrolling* fixes this issue (so disabling it might allow you to reproduce it).

**Comment by lydell** [comment]
I went to about:preferences#advanced and unchecked the "Smooth scrolling" checkbox. I still can't reproduce, though.

**Comment by progandy** [comment]
I can confirm that the <space>, <d>, and <u> sometimes scroll double the amount they should if smooth scrolling is disabled.
When I start Firefox, everything works as expected. I often use only standby and keep firefox running, so after 2-3 days, suddenly <d> and <u> scroll a whole page instead of half and <space> scrolls two pages. If i press <space> in the ignore mode, then it scrolls only one page.
If I enable smooth scrolling, everything is working without problems.

**Comment by progandy** [comment]
Hmm, I just solved it by deactivating and reactivating VimFx. Maybe there happens something strnge so that VimFx receives the command twice. I just saw that my VimFx copy was updated yesterday, so maybe the restartless update isn't perfectly clean. I don't remember if there were updates when it happened the other times.

**Comment by lydell** [comment]
@progandy How can you now that you "solved" it? It hasn't passed 2–3 days yet?

**Comment by now** [comment]
Listen, we’ve all been there, @lydell.  We’re frustrated or annoyed and we lose our temper and it’s easy to drop a few harshly worded sentences online to try to relieve the pressure.  But please, don’t do so again, because you’re doing so at the expense of others.

I saw an issue, figured others would have seen it as well and waited around for a while.  After a couple of days I checked in here to see if anyone had reported this issue.  No one had, so I did.  And after a while someone else came around and reported to be having the same issue.  Neither of us have asked you to do anything.  And we both found a solution that works around the issue.  Anyone having the same issue can now refer to this issue to try out that solution.

If there’s nothing obviously wrong with the page-scrolling code, then fine, leave this laying around until, well, whenever, I certainly don’t mind enabling Smooth Scrolling.

I can add that I don’t believe that I saw this issue before the 0.17.* series (and perhaps not even before 0.17.0).

**Comment by progandy** [comment]
@lydell I sent my first message when I noticed the problem, but without restarting Firefox. I temporarily enabled Smooth Scrolling, but didn't like it so I searched for another way to get the right behaviour back. A disable/enable cycle of vimfx worked without a firefox restart. It may not be a proper fix, but it is good enough for now.

**Comment by lydell** [comment]
> Maybe the problem will emerge again after a few days or maybe it comes from the standby cycles, we will see.

Yes, be on the lookout for it! Sooner or later I’m sure you’ll figure out how to reproduce and we’ll re-open. :) For now, however, I’m afraid there’s nothing to do.

**Comment by progandy** [comment]
I seem to have narrowed the problem down for me. It happens only when VimFx gets updated automatically. I cannot reproduce it with a reinstallation of the same vimfx version. 
I tried reproducing it in a new profile without success. I installed VimFx 0.17.3, disabled general.smoothScroll, restarted firefox, then chose to update all addons (i.e. to VimFx 0.17.4). After the update everything still works.
Maybe there is some conflict during the update process with some of the other addons I use like Stylish or Greasemonkey or uBlock. My addon list is just too long. Or maybe my profile is corrupted, and I should use a new one.
